rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlDocument.java

```diff
@@ -0,0 +1,149 @@
+
+    @Option(displayName = "Key path",
+            description = "A [JsonPath](https://docs.openrewrite.org/reference/jsonpath-and-jsonpathmatcher-reference) expression used to find matching keys.",
+            example = "$.metadata")
+    String key;
+
+    @Option(displayName = "YAML snippet",
+            description = "The YAML snippet to insert. The snippet will be indented to match the style of its surroundings.",
+            example = "labels:\n  label-one: \"value-one\"")
+    @Language("yml")
+    String yaml;
+
+    @Option(displayName = "Accept theirs",
+            description = "When the YAML snippet to insert conflicts with an existing key value pair and an existing key has a different value, prefer the original value.",
+            required = false)
+    @Nullable
+    Boolean acceptTheirs;
+
+    @Incubating(since = "7.30.0")
+    @Option(displayName = "Object identifying property",
+            description = "Name of a property which will be used to identify objects (mapping). This serves as the key to match on when merging entries of a sequence.",
+            required = false,
+            example = "name")
+    @Nullable
+    String objectIdentifyingProperty;
+
+    @Option(displayName = "Insert mode",
+            description = "Choose an insertion point when multiple mappings exist. Default is `Last`.",
+            valid = {"Before", "After", "Last"},
+            required = false)
+    @Nullable
+    InsertMode insertMode;
+
+    @Option(displayName = "Insert property",
+            description = "Define the key for the insertion mode. Takes the `key` JsonPath into account. Only useful when `insert mode` is either `Before` or `After`.",
+            required = false,
+            example = "some-key")
+    @Nullable
+    String insertProperty;
+
+    @Option(displayName = "Create new keys",
+            description = "When the key path does _not_ match any keys, create new keys on the spot. Default is `true`.",
+            required = false)
+    @Nullable
+    Boolean createNewKeys;
+
+    public MergeYamlDocument(String matcher, String key, @Language("yml") String yaml, @Nullable Boolean acceptTheirs, @Nullable String objectIdentifyingProperty, @Nullable InsertMode insertMode, @Nullable String insertProperty, @Nullable Boolean createNewKeys) {
+        this.matcher = matcher;
+        this.key = key;
+        this.yaml = yaml;
+        this.acceptTheirs = acceptTheirs;
+        this.objectIdentifyingProperty = objectIdentifyingProperty;
+        this.insertMode = insertMode;
+        this.insertProperty = insertProperty;
+        this.createNewKeys = createNewKeys;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Merge YAML snippet";
+    }
+
+    @Override
+    public String getInstanceNameSuffix() {
+        return String.format("at `%s`", key);
+    }
+
+    @Override
+    public String getDescription() {
+        return "Merge a YAML snippet with an existing YAML document.";
```

Comment:
Perhaps in `MergeYaml` this should be `Merge a YAML snippet with an existing YAML file.` I don't know if that semantic difference is clear to the user though